File: app/src/main/java/me/imzack/app/end/injector/module/MyPlansPresenterModule.java
package me.imzack.app.end.injector.module;

import me.imzack.app.end.view.contract.MyPlansViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class MyPlansPresenterModule {

    private final MyPlansViewContract mMyPlansViewContract;

    public MyPlansPresenterModule(MyPlansViewContract myPlansViewContract) {
        mMyPlansViewContract = myPlansViewContract;
    }

    @Provides
    MyPlansViewContract provideMyPlansViewContract() {
        return mMyPlansViewContract;
    }
}

File: app/src/main/kotlin/me/imzack/app/end/injector/module/MyPlansPresenterModule.kt
package me.imzack.app.end.injector.module

import dagger.Module
import dagger.Provides
import me.imzack.app.end.view.contract.MyPlansViewContract

@Module
class MyPlansPresenterModule(private val mMyPlansViewContract: MyPlansViewContract) {

    @Provides
    fun provideMyPlansViewContract() = mMyPlansViewContract
}
